MockitoJUnitRunnerToExtension does not handle MockitoJUnitRunner.StrictStubs, leaving @RunWith in place

### What version are you using?

Reproduced against current `main` (`3.44.0-SNAPSHOT`).

### What is the smallest, simplest way to reproduce the problem?

`MockitoJUnitRunnerToExtension` makes no change at all to a class using `MockitoJUnitRunner.StrictStubs`:

```java
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.StrictStubs.class)
public class ExternalAPIServiceTest {
}
```

Added to `MockitoJUnitRunnerToExtensionTest`, this fails with `java.lang.AssertionError: Recipe was expected to make a change but made no changes.`

### What did you expect to see?

```java
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
public class ExternalAPIServiceTest {
}
```

No explicit `@MockitoSettings` is needed: `javap -v` on `MockitoSettings.class` shows `AnnotationDefault: Lorg/mockito/quality/Strictness;.STRICT_STUBS`, and `.StrictStubs` is the one JUnit 4 runner whose behaviour matches that exactly. Verified at runtime (Mockito 5.12.0, JUnit 4.13.2, Jupiter 5.10.2, JDK 21):

| scenario | `@RunWith(...StrictStubs.class)` | bare `@ExtendWith(MockitoExtension.class)` |
|---|---|---|
| stub never invoked | FAIL `UnnecessaryStubbingException` | FAIL `UnnecessaryStubbingException` |
| cross-file arg mismatch, stub never matched | FAIL `PotentialStubbingProblem` | FAIL `PotentialStubbingProblem` |
| cross-file mismatch while stub unused, then a matching call | FAIL `PotentialStubbingProblem` | FAIL `PotentialStubbingProblem` |

(Reproducing `PotentialStubbingProblem` requires the stubbing and the mismatched invocation to be in *different source files* — `DefaultStubbingLookupListener.potentialArgMismatches` compares source files. A single-file repro will not show it.)

This is also the exact shape the recipe already emits for its `STRICT_STUBS` mapping, where `getTemplate()` returns `Optional.empty()`.

### What did you see instead?

No change whatsoever — `@RunWith(MockitoJUnitRunner.StrictStubs.class)` is left in place.

### Root cause

`MockitoJUnitRunner.StrictStubs` appears nowhere in `src/main`. It is missing from both:

- the `Strictness` enum — `MockitoJUnitRunnerToExtension.java:143-145`
- the `obsoleteRunners` list — `:111-114`

The `UsesType<>("org.mockito.junit.MockitoJUnitRunner*", false)` precondition at `:56` *does* match `org.mockito.junit.MockitoJUnitRunner$StrictStubs`, so the visitor does run. The blocker is `getStrictness()` (`:85-103`):

- `TypeUtils.isOfClassType` is name-exact (it treats `$` and `.` as equivalent), not assignability-based — so even though `MockitoJUnitRunner$StrictStubs extends MockitoJUnitRunner`, the `WARN` entry does not match.
- `visitFieldAccess` (`:89-101`) returns without calling `super`, so the nested `FieldAccess` — whose target *is* the plain `MockitoJUnitRunner` and would have matched `WARN` — is never visited.

All comparisons return `false`, `getStrictness()` returns `null`, and `visitClassDeclaration` returns at `:69-71` before reaching `registerAfterVisit`.

Nothing else in the chain covers it either. `MockitoJUnitToMockitoExtension` is precondition-gated on `MockitoRule` / `MockitoTestRule` and its matcher targets the pre-2.x `org.mockito.runners.MockitoJUnitRunner`; `RemoveObsoleteRunners` in `junit5.yml:63-66` lists only `org.junit.runners.JUnit4` and `org.junit.runners.BlockJUnit4ClassRunner`.

### Consequence: non-compiling output under `JUnit4to5Migration`

Run on its own, `MockitoJUnitRunnerToExtension` (or `UseMockitoExtension`) leaves the class unmigrated but still compiling.

Under `JUnit4to5Migration` — and therefore `JUnit5BestPractices`, which includes it — the result does not compile, because a single `recipeList` contains both the Mockito migration and the JUnit 4 dependency removal:

```
org.openrewrite.java.testing.junit5.JUnit4to5Migration      (junit5.yml:46)
  ...
  - org.openrewrite.java.testing.junit5.UseMockitoExtension  (junit5.yml:100)
  - org.openrewrite.java.dependencies.RemoveDependency:      (junit5.yml:103)
      groupId: junit
      artifactId: junit                                      (junit5.yml:105)
```

Confirmed end-to-end: running `JUnit4to5Migration` over the class above together with a pom declaring `junit:junit:4.13.2` leaves the Java source byte-identical while `junit:junit` is removed from the pom. `import org.junit.runner.RunWith;` and the annotation survive, now referencing a dependency that is no longer on the classpath.

### Note on fixing it

This is not a one-line enum addition. `getTemplate()` interpolates the enum constant name straight into the template (`:132`), so the constant names must be exactly the three `org.mockito.quality.Strictness` values — a fourth constant is not possible, and the `STRICT_STUBS` slot is currently held by `MockitoJUnitRunner.Strict`. So the fix needs either a `Strictness` restructured to associate several runner FQNs with one level, or a remapping of `.Strict`.

Both `obsoleteRunners` (`:111-114`) and the enum (`:143-145`) need the new entry — a patch touching only the enum will compute the right strictness and still leave `@RunWith` in place.

For context, and explicitly *not* something I'm proposing to change here: `MockitoJUnitRunner.Strict` and the base `MockitoJUnitRunner` have byte-for-byte identical constructors in Mockito 5.12.0 (both `StrictRunner(new RunnerFactory().createStrict(klass), klass)`, installing `MismatchReportingTestListener`), yet are mapped to different levels. I looked at remapping `.Strict` to free the slot and concluded it would be a regression: `.Strict` → `STRICT_STUBS` diverges from `.Strict`'s measured behaviour in fewer scenarios than `.Strict` → `WARN` would, and the current output is published documentation in `examples.yml`. Happy to discuss separately if useful.

I have a failing test ready and would be glad to open a PR.

--- a/src/main/java/org/openrewrite/java/testing/mockito/MockitoJUnitRunnerToExtension.java
+++ b/src/main/java/org/openrewrite/java/testing/mockito/MockitoJUnitRunnerToExtension.java
@@ -15,9 +15,7 @@
  */
 package org.openrewrite.java.testing.mockito;
 
-import lombok.AccessLevel;
 import lombok.Getter;
-import lombok.RequiredArgsConstructor;
 import org.jspecify.annotations.Nullable;
 import org.openrewrite.ExecutionContext;
 import org.openrewrite.Preconditions;
@@ -88,7 +86,7 @@ public class MockitoJUnitRunnerToExtension extends Recipe {
                         @Override
                         public J.FieldAccess visitFieldAccess(J.FieldAccess fieldAccess, AtomicReference<Strictness> strictness) {
                             for (Strictness strict : Strictness.values()) {
-                                if (TypeUtils.isOfClassType(fieldAccess.getTarget().getType(), strict.runner)) {
+                                if (strict.runners.stream().anyMatch(runner -> TypeUtils.isOfClassType(fieldAccess.getTarget().getType(), runner))) {
                                     strictness.set(strict);
                                     break;
                                 }
@@ -111,6 +109,7 @@ public class MockitoJUnitRunnerToExtension extends Recipe {
                 List<String> obsoleteRunners = Arrays.asList(
                         "org.mockito.junit.MockitoJUnitRunner.Silent",
                         "org.mockito.junit.MockitoJUnitRunner.Strict",
+                        "org.mockito.junit.MockitoJUnitRunner.StrictStubs",
                         "org.mockito.junit.MockitoJUnitRunner");
                 if (hasMockitoExtensions) {
                     doAfterVisit(new RemoveObsoleteRunners(obsoleteRunners).getVisitor());
@@ -118,7 +117,7 @@ public class MockitoJUnitRunnerToExtension extends Recipe {
                     doAfterVisit(new RunnerToExtension(obsoleteRunners, "org.mockito.junit.jupiter.MockitoExtension").getVisitor());
                 }
                 for (Strictness strictness : Strictness.values()) {
-                    maybeRemoveImport(strictness.runner);
+                    strictness.runners.forEach(this::maybeRemoveImport);
                 }
                 maybeAddImport("org.mockito.quality.Strictness");
                 maybeAddImport("org.mockito.junit.jupiter.MockitoSettings");
@@ -138,13 +137,16 @@ public class MockitoJUnitRunnerToExtension extends Recipe {
         });
     }
 
-    @RequiredArgsConstructor(access = AccessLevel.PACKAGE)
     private enum Strictness {
         LENIENT("org.mockito.junit.MockitoJUnitRunner.Silent"),
         WARN("org.mockito.junit.MockitoJUnitRunner"),
-        STRICT_STUBS("org.mockito.junit.MockitoJUnitRunner.Strict");
+        STRICT_STUBS("org.mockito.junit.MockitoJUnitRunner.Strict", "org.mockito.junit.MockitoJUnitRunner.StrictStubs");
 
-        final String runner;
+        final List<String> runners;
+
+        Strictness(String... runners) {
+            this.runners = Arrays.asList(runners);
+        }
 
         // Return true, if current strictness is greater than given strictness.
         boolean isGreaterThan(Strictness strictness) {

--- a/src/test/java/org/openrewrite/java/testing/mockito/MockitoJUnitRunnerToExtensionTest.java
+++ b/src/test/java/org/openrewrite/java/testing/mockito/MockitoJUnitRunnerToExtensionTest.java
@@ -66,6 +66,31 @@ class MockitoJUnitRunnerToExtensionTest implements RewriteTest {
         );
     }
 
+    @Test
+    void strictStubsMockitoRunnerToExtension() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.runner.RunWith;
+              import org.mockito.junit.MockitoJUnitRunner;
+
+              @RunWith(MockitoJUnitRunner.StrictStubs.class)
+              public class ExternalAPIServiceTest {
+              }
+              """,
+            """
+              import org.junit.jupiter.api.extension.ExtendWith;
+              import org.mockito.junit.jupiter.MockitoExtension;
+
+              @ExtendWith(MockitoExtension.class)
+              public class ExternalAPIServiceTest {
+              }
+              """
+          )
+        );
+    }
+
     @CsvSource({
       "MockitoJUnitRunner.Silent.class,Strictness.LENIENT",
       "MockitoJUnitRunner.class,Strictness.WARN"

